### ISSUE TITLE
a11y: Make comment accessible in channels tree

### ISSUE THREAD

**Issue by jcsteh** [mumble-voip/mumble#2292]
The comment for a user appears in a tool tip when you mouse over that user in the channels tree. Unfortunately, this is not easily accessible to users of screen readers. It would be good if this could be exposed as the accessible description for the tree item.

I have no experience with QT programming, but glancing at the API and the Mumble code, I believe you should be able to manage this by exposing the comment via the [Qt::AccessibleDescriptionRole](http://doc.qt.io/qt-5/qt.html#ItemDataRole-enum).

**Comment by mkrautz** [comment]
I'm not sure it's enough to just expose the comment as the AcessibleDescriptionRole, but I suppose it would be a start!

Something like a new dialog box that contains information about a user like:

- Their name
- Their current comment
- Their status (muted, deafened, self-muted, self-deafened, suppressed)

sounds like a better approach to me. I'm not sure if it really is, though.

Or a "View Comment" context menu, maybe.

**Comment by jcsteh** [comment]
A dialog with all of that info would be nice. However, that alone would 
make for a fairly inefficient UX for a screen reader user, as they'd 
have to open the dialog just to check whether a user even has a comment. 
That's much faster for a sighted user; they can just mouse over the 
user, and as I understand it, there's even an icon indicating that the 
user has a comment.

Let me ask this question: what states have visual indication in the 
tree? Are there icons for all of those (comment, muted, deafened, etc.)? 
If so, we probably want to expose at least some of those in the 
accessible description as well. The key point is that it'd be good if 
the experience were just as efficient for a screen reader user.

**Comment by mkrautz** [comment]
Yes, all of those are represented by icons.

My worry is that a comment is user-defined, and can be quite long.

That is why I thought a dialog would be better: a user using a screen reader could simply move focus to the comment text, and if it was too long... well, change focus to the next item.

If everything is put into Qt::AcessibleDescriptionRole, it could be quite a long text.

Maybe a better solution would be to make the icons representing the user's state (comment, muted, deafned, self-muted, self-deafened, suppressed, authenticated) be focusable. In the general case, th

**Comment by jcsteh** [comment]
It's certainly true that a long comment could make things a bit ugly. On the flip side, it can also be used to indicate status (e.g. "AFK" or "lunch"), and in that case, being able to get it as quickly as possible is very useful. Having to specifically focus the icons still means there has to be some special interaction (other than moving to the user in the tree) to get that information. To give you an example use case, imagine you have a few users in a channel and you want to quickly know whether they are around. With the accessible description idea, you could just cursor through the tree and get a quick summary. If you had to focus each icon, it'd certainly give you more granularity, but it's also less efficient/more tedious. Note that a screen reader user can choose to interrupt reading if they're not interested; you just want to make sure less important information comes later.

In summary, I suggest the following:

1. Expose a summary of the information in the accessible description with the comment being last. So, this might be "muted deafened, Eating lunch", where "Eating lunch" is the comment. If you're really worried about the length of the comment, you could truncate it with an ellipsis at a certain length.
2. Being able to inspect this information separately in a dialog might still be useful in some cases. However, this certainly means more implementation effort and I personally think it's far lower priority.
